RePak/src/assets/rui.cpp

```diff
@@ -1,15 +1,60 @@
+    // JSON VALIDATION
+    {
+        // atlas checks
+        if (!mapEntry.HasMember("atlas"))
+            Error("Required field 'atlas' not found for uimg asset '%s'. Exiting...\n", assetPath);
+        else if (!mapEntry["atlas"].IsString())
+            Error("'atlas' field is not of required type 'string' for uimg asset '%s'. Exiting...\n", assetPath);
+
+        // textures checks
+        if (!mapEntry.HasMember("textures"))
+            Error("Required field 'textures' not found for uimg asset '%s'. Exiting...\n", assetPath);
+        else if (!mapEntry["textures"].IsArray())
+            Error("'textures' field is not of required type 'array' for uimg asset '%s'. Exiting...\n", assetPath);
+
+        // validate fields for each texture
+        for (auto& it : mapEntry["textures"].GetArray())
+        {
+            if (!it.HasMember("path"))
+                Error("Required field 'path' not found for a texture in uimg asset '%s'. Exiting...\n", assetPath);
+            else if (!it["path"].IsString())
+                Error("'path' field is not of required type 'string' for a texture in uimg asset '%s'. Exiting...\n", assetPath);
+
+            if (!it.HasMember("width"))
+                Error("Required field 'width' not found for a texture in uimg asset '%s'. Exiting...\n", assetPath);
+            // technically this could be a float i think? Going to limit it to ints for now though for simplicity
+            else if (!it["width"].IsNumber())
+                Error("'width' field is not of required type 'number' for a texture in uimg asset '%s'. Exiting...\n", assetPath);
+
+            if (!it.HasMember("height"))
+                Error("Required field 'height' not found for a texture in uimg asset '%s'. Exiting...\n", assetPath);
+            // technically this could be a float i think? Going to limit it to ints for now though for simplicity
+            else if (!it["height"].IsNumber())
+                Error("'height' field is not of required type 'number' for a texture in uimg asset '%s'. Exiting...\n", assetPath);
+
+            if (!it.HasMember("posX"))
+                Error("Required field 'posX' not found for a texture in uimg asset '%s'. Exiting...\n", assetPath);
+            // technically this could be a float i think? Going to limit it to ints for now though for simplicity
+            else if (!it["posX"].IsNumber())
+                Error("'posX' field is not of required type 'number' for a texture in uimg asset '%s'. Exiting...\n", assetPath);
+
+            if (!it.HasMember("posY"))
+                Error("Required field 'posY' not found for a texture in uimg asset '%s'. Exiting...\n", assetPath);
+            // technically this could be a float i think? Going to limit it to ints for now though for simplicity
+            else if (!it["posY"].IsNumber())
+                Error("'posY' field is not of required type 'number' for a texture in uimg asset '%s'. Exiting...\n", assetPath);
```

Comment:
at this point, the path has already been validated and can be used for error messages.
e.g.
instead of "for a texture", you could use: "for texture '%s'", using the checked path as input